src/lib/network/edit_nic_name.rb

```diff
@@ -68,11 +74,16 @@ def run
         # FIXME: it changes udev key used for device identification
         #  and / or its value only, name is changed elsewhere
         LanItems.update_item_udev_rule!(udev_type)
+
+        if new_name != old_name
```

Comment:
we already have ```LanItems.renamed?```, should be usable here.

Reply:
It is not enough... Use case:

1. Modify Item name & update routing devices. (renamed? true) eth0 -> em0
2. Modify Item name from em0 -> eth0 (renamed? false, but routes should be updated)

Reply:
well in fact I think that updating routes after each name change is not needed. It should be done when user actually writes the configuration. Not when he fills in new name.

Reply:
I originally added also when "Committing" changes, then realized that "Cancel" also doesn't revert name changes, so moved the responsibility close to the action where persistent changes (of course before final write) where done. But permitting the user to decide if update them or not. 

So, about when **Write** the configuration, that is exactly what reported. If you modify the name and goes to the Routing section the references will be still there.

Reply:
https://github.com/yast/yast-network/pull/725/commits/40d4aebeb0971aa1a3aff4c5033deabd57684d54#diff-72293d1fba24e0701afaa84e81e1af12R1394

Reply:
then we have a bug that "Cancel" do not revert name change, which should be fixed.

Reply:
> well in fact I think that updating routes after each name change is not needed. It should be done when user actually writes the configuration. Not when he fills in new name.

So, what do you propose?

Reply:
> then we have a bug that "Cancel" do not revert name change, which should be fixed.

Yep, probably Udev name rules are not reverted when canceling, so basically another missing callback.

Reply:
I propose to fix cancel behavior instead of introducing new exceptional behavior.

Reply:
Fine with me, but then propose to do it separately, in other PR as that is another bug.

Reply:
no problem, then pls create a bug and link this PR to make future fix of this behavior easier.

Reply:
I could move the logic back to the Address dialog method, but I preferred to have it Isolated in the change nic name dialog until we fix what already discussed.

As I said, it is strange to ask about updating routes when canceling, so, it is not so strange to ask when changing name, then you have the option to not confirm and cancel, of course it is still not perfect but better than crashing.